src/test/java/com/flightstats/hub/util/IntegrationUdpServer.java

```diff
@@ -17,57 +17,66 @@
 import java.net.DatagramPacket;
 import java.net.DatagramSocket;
 import java.io.IOException;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.stream.Collectors;
 
 @Builder
 public class IntegrationUdpServer {
     private final static Logger logger = LoggerFactory.getLogger(IntegrationUdpServer.class);
-    private long timeoutMillis;
-    private final List<String> queue = new ArrayList<>();
     private int port;
-    private boolean listening;
+    private AtomicBoolean listening;
+    private CountDownLatch startupCountDownLatch;
+    private AtomicBoolean canListen;
+    private ExecutorService executorService;
 
     private final Map<String, String> store = new HashMap<>();
 
-    private final CompletableFuture<Map<String, String>> serverFuture = CompletableFuture.supplyAsync(() -> {
-        try {
-            DatagramSocket serverSocket = new DatagramSocket(port);
-            while (listening) {
-                byte[] data = new byte[70];
-                String result = openSocket(data, serverSocket);
-                addValueToStore(result);
+    public CompletableFuture<Map<String, String>> getServerFuture() {
+        return  CompletableFuture.supplyAsync(() -> {
+            try {
+                DatagramSocket serverSocket = new DatagramSocket(port);
+                while (listening.get()) {
+                    byte[] data = new byte[70];
+                    DatagramPacket receivePacket = new DatagramPacket(data, data.length);
+                    startupCountDownLatch.countDown();
+                    serverSocket.receive(receivePacket);
+
+                    String result = listen(receivePacket);
+                    addValueToStore(result);
+                }
+            } catch (IOException e) {
+                listening.set(false);
             }
-        } catch(Exception ex) {
-            listening = false;
-        }
 //        logger.info(":::::::::, {}", store);
```

Comment:
why is this commented out?

Reply:
Because it was originally and I didn't change it. I can remove it.